packages/vuetify/src/mixins/overlayable/index.ts

```diff
@@ -146,19 +136,29 @@ export default Vue.extend<Vue & Toggleable & Stackable & options>().extend({
         }
       }
 
-      if (e.target === this.overlay ||
-        (e.type !== 'keydown' && e.target === document.body) ||
-        this.checkPath(e)) e.preventDefault()
+      if (e.target === this.overlay || (e.type !== 'keydown' && e.target === document.body) || this.checkPath(e)) e.preventDefault()
     },
     hasScrollbar (el?: Element) {
       if (!el || el.nodeType !== Node.ELEMENT_NODE) return false
 
       const style = window.getComputedStyle(el)
       return ['auto', 'scroll'].includes(style.overflowY!) && el.scrollHeight > el.clientHeight
     },
-    shouldScroll (el: Element, delta: number) {
-      if (el.scrollTop === 0 && delta < 0) return true
-      return el.scrollTop + el.clientHeight === el.scrollHeight && delta > 0
+    shouldScroll (this: any, el: Element, delta: number) {
+      const isAppWrapper = el.classList.contains('v-application')
+
+      if (isAppWrapper) return false
+
+      const alreadyAtTop = el.scrollTop === 0
+      const alreadyAtBottom = el.scrollTop + el.clientHeight === el.scrollHeight
```

Comment:
This seems like it might be pretty expensive to do recursively on every scroll event?

Reply:
It's been working alright for me, but yes if you have tons of nested content it could be slow...but since we're overriding default scroll behavior I'm not sure how else to bubble the scroll event up through parents to mimic the default behavior we lost.

I'm definitely open to suggestions, though, if there is a more performant way to do this, or a way to skip elements that don't have the ability to scroll/are not overflowing content.